src/dotnet/APIView/APIViewWeb/Languages/CLanguageService.cs

```diff
@@ -82,7 +84,12 @@ public override async Task<CodeFile> GetCodeFileAsync(string originalName, Strea
             var archive = new ZipArchive(astStream);
 
             //Generate pacakge name from original file name
-            var packageNamespace = originalName.Substring(0, originalName.LastIndexOf(".")).Replace("_", ".");
+            string packageNamespace = "";
+            var packageNameMatch = _packageNameParser.Match(originalName);
+            if (packageNameMatch.Success)
+            {
+                packageNamespace = packageNameMatch.Groups[1].Value.Replace("_", "::");
```

Comment:
Does C have namespaces?

Reply:
It doesn't. I am using this only to set package name to have a uniform way to find review automatically. It's not used to filter within review.